src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -206,7 +206,7 @@ public void Message(string Message)
         /// Events from DiagnosticSource can be forwarded to EventSource using this event.
         /// </summary>
         [Event(2, Keywords = Keywords.Events)]
-        private void Event(string SourceName, string EventName, IEnumerable<KeyValuePair<string, string>> Arguments)
+        private void Event(string? SourceName, string EventName, IEnumerable<KeyValuePair<string, string?>>? Arguments)
```

Comment:
I wouldn't have expected SourceName to be nullable. Did you find a case where this occurs?

(As a private member I think its a low priority to get this annotation correct, so not overly concerned either way)

Reply:
> I wouldn't have expected SourceName to be nullable. Did you find a case where this occurs?

i didn't found exact call with null, nor found the case it preventing null input (i.e. throw on null or so). So it can be both, nullable or not nullable. The  method called inside this method accepts null values `protected void WriteEvent(int eventId, string? arg1, string? arg2, string? arg3);` and an Action delegate accepts all nullable. (These methods used for action implementation). So it evaluated to can be null.

> (As a private member I think its a low priority to get this annotation correct, so not overly concerned either way)

Yes its nullability not affecting public API and doesn't affect compiler, i didn't annotated similar methods below as i noticed its private and not affecting anything. But i would make it non null for convenience